core/src/main/java/io/seata/core/store/db/LockStoreDataBaseDAO.java

```diff
@@ -114,25 +120,29 @@ public boolean acquireLock(List<LockDO> lockDOs) {
             }
             rs = ps.executeQuery();
             while (rs.next()) {
-                if (StringUtils.equals(rs.getString("xid"), lockDOs.get(0).getXid())) {
-                    isReLock = true;
-                } else {
+                if (!StringUtils.equals(rs.getString("xid"), lockDOs.get(0).getXid())) {
                     canLock &= false;
+                    break;
```

Comment:
why break?
db: lock1:xid1   lock2:xid2  lock3:xid1
request: lock1:xid1  lock2:xid1  lock3:xid1

lock3 can add in dbExistedRowKeys?

Reply:
db: lock2:xid2
request:  lock2:xid1
These two global transactions are different. You cannot get the lock this time.
add break is a quick failure

Reply:
agree